Add Score Progress bar to Roses

danleedham opened the pull request:
I think I've finally built something that's finished enough to eb added. 

User Toggles on Winning Progress which presents a bar on the left hand side . 

Displays nothing when at zero, but as we'll be streaming after some points have already scored that's academic. 

Works will josn and manual scores. 

![screen shot 2018-04-10 at 20 10 40](https://user-images.githubusercontent.com/12582998/38578249-498c405e-3cfb-11e8-9321-c7836394a8f8.png)

Tested on Caspar Server Beta 2.1.0. Not tested with specific video underneath so might want to change the color of the the 177.5, or pop it in a box etc.

danleedham committed:
Adds progress bar option to Roses scores

This displays a current score progress bar on the left hand side with pretty Roses.

danleedham committed:
Change json back from testing to live

danleedham committed:
Fixed 177.5 not being on the line

stephenrob commented on public/js/cg.js:
```diff
@@ -140,14 +140,22 @@ app.controller('scoringCtrl', ['$scope', '$interval', '$http', 'socket',
             };
+            
+            var available = $scope.yorkScore + $scope.lancScore;
+			var toWin = 177.5;
```
Could we extract this out to a constant at the top making this easier to change? Might also be worth making note of the total points available as well.

stephenrob reviewed (changes requested):
Location of 'points to win' definition to be clearer and easy to edit.

danleedham commented:
Is the the points to win likely to change across the weekend? 

I had thought of using the fixtures data to count all the points available, then do the math but was worried that could lead to errors if the data wasn't perfect. 

The available variable doesn't actually do anything, it was just if I wanted to use the current amount of points currently scored. I'll remove it for the absence of doubt. 

Is the total points available not always `(toWin x2 ) - 1` ?

danleedham committed:
Make score to win part of the scope, place at top of controller

danleedham committed:
Make Roses available points dynamic

Adds backend option for changing the total number of points available.

This then trundles through the code and updates the graphic.

stephenrob commented:
We can look at possibly pulling the total points available from roseslive once i've evaluated the code but for now this can be merged in.
